hazelcast-jet-core/src/main/java/com/hazelcast/jet/processor/Sources.java

```diff
@@ -61,9 +64,33 @@ public static ProcessorMetaSupplier readMap(@Nonnull String mapName) {
     }
 
     /**
+     * Returns a meta-supplier of processor that will fetch entries from the
+     * Hazelcast {@code IMap} with the specified name. Entries will be filtered
+     * according to the given predicate and the result of the projection will be
+     * emitted to downstream. The processors will only access data local to the
+     * member and, if {@code localParallelism} for the vertex is above one,
+     * processors will divide the labor within the member so that each one gets
+     * a subset of all local partitions to read.
+     * <p>
+     * The number of Hazelcast partitions should be configured to at least
+     * {@code localParallelism * clusterSize}, otherwise some processors will have
+     * no partitions assigned to them.
+     * <p>
+     * If the underlying map is concurrently being modified, there are no guarantees
+     * given with respect to missing or duplicate items.
+     */
+    @Nonnull
+    public static <K, V, T> ProcessorMetaSupplier readMap(@Nonnull String mapName,
+                                                          @Nonnull DistributedFunction<Map.Entry<K, V>, T> projectionF,
```

Comment:
We have similar concept here: https://github.com/hazelcast/hazelcast-jet/blob/master/hazelcast-jet-core/src/main/java/com/hazelcast/jet/processor/DiagnosticProcessors.java#L94-L97
You argued that requiring @Nonnull is better, but I don't like it at all. It's much easier to write `null` to predicate than `() -> true`.